### ISSUE TITLE
Add option to "sms via voip.ms" to contact book entries

### ISSUE THREAD

**Issue by singpolyma** [michaelkourlas/voipms-sms-client#110]
Similar to "call via whatsapp" buttons and similar. So a new sms conversation can be started in this app via the normal contacts.

**Comment by michaelkourlas** [comment]
Unfortunately VoIP.ms SMS isn't an SMS app from the OS's point of view, so that's not possible. See issue #49 for details.

**Comment by singpolyma** [comment]
Pretty sure any app can add a contact option to a contact. Doesn't have anything to do with SMS...?

**Comment by michaelkourlas** [comment]
To clarify, you're asking for the ability to click a button in your contact list that opens the conversation window for that contact in VoIP.ms SMS?

**Comment by singpolyma** [comment]
yes

**Comment by michaelkourlas** [comment]
Yes, that requires that VoIP.ms SMS advertise the SEND_TO intent, which isn't possible for SMS apps unless they're the default.

**Comment by singpolyma** [comment]
but why do you need to say you're an sms app? why not just be the same as any other messaging app that doesn't do sms?

**Comment by michaelkourlas** [comment]
What other messaging apps let you launch them from your contacts?

**Comment by singpolyma** [comment]
See for example http://stackoverflow.com/questions/28583233/special-case-when-adding-a-custom-connection-action-to-contacts-in-android-kitka and option in screnshot to skype with a username (not phone number)

**Comment by michaelkourlas** [comment]
Fair enough. This is something different then what I thought you were talking about, and worth investigating.

**Comment by michaelkourlas** [comment]
Additional related link: https://stackoverflow.com/questions/29846912/adding-new-connection-to-existing-android-contact

**Comment by tareko** [comment]
Any further thoughts on this? It is a very minor irritation, as VoIP.ms is my main number.

I'm not an android programmer, but if I can contribute in any other way, please let me know.

Tarek : )

**Comment by michaelkourlas** [comment]
https://developer.android.com/guide/topics/providers/contacts-provider
